tests/php/Control/Middleware/RewriteHashLinksMiddlewareTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace SilverStripe\Control\Tests\Middleware;
+
+use SilverStripe\Control\Middleware\RewriteHashLinksMiddleware;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Convert;
+use SilverStripe\Dev\SapphireTest;
+
+class RewriteHashLinksMiddlewareTest extends SapphireTest
+{
+    /**
+     * Tests rewriting of HTML content types to ensure that it's properly rewriting anchors when the base tag is present
+     */
+    public function testRewriteHTMLWithBase()
+    {
+        $_SERVER['HTTP_HOST'] = 'www.mysite.com';
```

Comment:
Hmmm only problem is that especially with HTTP_HOST there's no other way to set that in the HTTPRequest. What do you think about backing up the values during test setup and tear down?

Reply:
I'm pretty sure phpunit already backs up the globals. If it doesn't by default it can be enabled

Reply:
It looks like SapphireTest actually turns backup globals off "to avoid overriding the few globals SilverStripe relies on, like $lang for the i18n subsystem". I'll just implement something that backs up the host and request uri and resets it.